src/wasm/wasm-type.cpp

```diff
@@ -96,7 +95,33 @@ struct HeapTypeInfo {
   HeapTypeInfo& operator=(const HeapTypeInfo& other);
   bool operator==(const HeapTypeInfo& other) const;
   bool operator!=(const HeapTypeInfo& other) const { return !(*this == other); }
-  bool operator<(const HeapTypeInfo& other) const;
+};
+
+struct TypeComparator {
+  std::unordered_set<std::pair<HeapType, HeapType>> cache;
+  bool cmp(Type a, Type b);
```

Comment:
Please add a comment explaining what the resulting bools here mean.

Lower down I see a call to this replaces `bool HeapType::operator<` so perhaps this could be `operator<` or be called `lessThan`?